compiler/ballerina-lang/src/main/java/org/ballerinalang/toml/parser/SettingsBuildListener.java

```diff
@@ -106,6 +107,11 @@ private void setToManifest(String value) {
      * @param key key specified in the header
      */
     private void addHeader(String key) {
-        currentHeader = key;
+        // Check if the header is valid for the Settings.toml
+        if ((SettingHeaders.valueOfLowerCase(key) != null)) {
```

Comment:
Inverse this if with else and fail faster

Reply:
Fixed with 8e24fff